internal/pkg/client/library/library.go

```diff
@@ -24,27 +24,37 @@ const defaultTag = "latest"
 // Comparison will be lower case as the GUI / code has used different capitalisation through time.
 const noDescription = "no description"
 
-// NormalizeLibraryRef strips off leading "library://" prefix, if any, and
-// appends the default tag (latest) if none specified.
-func NormalizeLibraryRef(libraryRef string) string {
-	ir := strings.TrimPrefix(libraryRef, "library://")
-	if !strings.Contains(ir, ":") {
-		return ir + ":" + defaultTag
-	}
-	return ir
-}
+// NormalizeLibraryRef parses library ref and sets default tag, if necessary.
+func NormalizeLibraryRef(ref string) (*scslibrary.Ref, error) {
+	ref = strings.TrimPrefix(ref, "library://")
 
-// DownloadImage is a helper function to wrap library image download operation
-func DownloadImage(ctx context.Context, c *scslibrary.Client, imagePath, arch, libraryRef string, callback client.ProgressCallback) error {
-	// reassemble "stripped" library ref for scs-library-client
-	validLibraryRef := "library:///" + libraryRef
+	slashes := strings.Count(ref, "/")
 
-	// parse library ref
-	r, err := scslibrary.Parse(validLibraryRef)
-	if err != nil {
-		return fmt.Errorf("error parsing library ref: %v", err)
+	var elem, tags []string
+	var host, pathref string
+
+	if slashes == 0 || slashes == 2 {
```

Comment:
I think this changes behavior? At present you can have slashes ==1 when you ommit the entity (in this example `library`) but specify `collection/container`

```
12:34 PM $ singularity pull library://default/ubuntu
INFO:    Downloading library image
743.6KiB / 26.8MiB [=>-----------------------------------------------------------------------] 3 % 413.1 KiB/s 1m4s
```